File: compiler/lua_sample.lua
var __extends = this.__extends || function (d, b) {
    function __() { this.constructor = d; }
    __.prototype = b.prototype;
    d.prototype = new __();
};
function afun() {
    var i, b;
    for(i = 1; 10; 2) {
        print(i);
        print(i);
    }
}
var M;
(function (M) {
    var Token;
    (function (Token) {
        Token._map = [];
        Token._map[0] = "ID";
        Token.ID = 0;
        Token._map[1] = "Param";
        Token.Param = 1;
        Token._map[2] = "Oper";
        Token.Oper = 2;
    })(Token || (Token = {}));
    var Parent = (function () {
        function Parent() { }
        Parent.prototype.say = function () {
            print("say");
        };
        Parent.prototype.test = function () {
        };
        Parent.prototype.why = function () {
        };
        return Parent;
    })();    
    var Child = (function (_super) {
        __extends(Child, _super);
        function Child(field, date) {
            supper();
            this.field = field;
            this.date = date;
        }
        Child.prototype.say = function () {
            print("say");
        };
        Child.prototype.test = function () {
        };
        return Child;
    })(Parent);    
})(M || (M = {}));
1 + 2 ~= 3;
function afun2() {
    for(i = 1; 10; 2) {
        print(i);
        print(i);
    }
}
if(1 + 1 == 2 and 1 + 2 >= 3) {
    print("true");
} else {
    print("false");
}
do {
    print("Hello");
}while(1 + 1 ~= 2)
while(1 + 1 ~= 2) {
    print("true");
}
var i = 5;
var a, b;
var c = 1, d = "a", e;

File: compiler/lua_sample.js
local i = 5
local a, b
local c, d, e = 1, "a"
function afun() 
    local i, b
    for i = 1, 10, 2 do 
        print(i);
        print(i);
    end 
end
(function (M) {
    (function (Token) {
        Token._map
        Token._map[0] = "ID";
        Token.ID
        Token._map[1] = "Param";
        Token.Param
        Token._map[2] = "Oper";
        Token.Oper
    })(M.Token || (M.Token = {}));
    var Token = M.Token;
    local Parent = Object:extend()
    function initialize:initialize(id) 
    end
    function Parent:say() 
        print(self.name);
        print(self.id);
    end
    function Parent:test() 
    end
    function Parent:why() 
    end

    M.Parent = Parent;    
    local Child = Parent:extend()
    function initialize:initialize(field, date) 
        _super.call(this, 2);
        self.field = field;
        self.date = date;
    end
    function Child:say() 
        print("say");
    end
    function Child:test() 
    end
Parent    
})(exports.M || (exports.M = {}));
var M = exports.M;
1 + 2 ~= 3;
function afun2() 
    for i = 1, 10, 2 do 
        print(i);
        print(i);
    end 
end
local ifa
if ifa == nil then 
    print(a);
end
if 1 + 1 == 2 and 1 + 2 >= 3 then 
    print("true");
elseif 1 + 2 ~= 3 then 
    print("true");
else 
    print("false");
end
repeat 
    print("Hello");
until 1 + 1 ~= 2
while 1 + 1 ~= 2 do 
    print("true");
end 
